behaviortree-rs/src/blackboard.rs

```diff
@@ -132,13 +133,21 @@ pub type EntryPtr = Arc<Mutex<Entry>>;
 
 impl Blackboard {
     fn new(parent_bb: Option<Blackboard>) -> Blackboard {
+        let root_bb = match &parent_bb {
+            Some(parent) => match *parent.root_bb.clone() {
+                Some(root) => Box::new(Some(root.clone())),
+                None => Box::new(parent_bb.clone()),
+            },
+            None => Box::new(None),
+        };
```

Comment:
This method will only go up one level in the hierarchy, so if `parent_bb` has two ancestors (a parent and a "grandparent"), then you will not get the "true" root blackboard. I think a better long-term approach to this will be to get to the root blackboard by traversing up the ancestry of the current blackboard every time, rather than storing a pointer to the root blackboard in each blackboard instance.